server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterStateBlobStore.java

```diff
@@ -88,18 +90,27 @@ public void readAsync(final U entity, final ActionListener<T> listener) {
         });
     }
 
-    private BlobPath getBlobPathForUpload(final AbstractRemoteWritableBlobEntity<T> obj) {
-        BlobPath blobPath = blobStoreRepository.basePath()
-            .add(RemoteClusterStateUtils.encodeString(clusterName))
-            .add("cluster-state")
-            .add(obj.clusterUUID());
+    public String getClusterName() {
+        return clusterName;
+    }
+
+    public BlobPath getBasePath() {
+        return blobStoreRepository.basePath();
+    }
+
+    public BlobPath getBlobPathPrefix(String clusterUUID) {
+        return getBasePath().add(RemoteClusterStateUtils.encodeString(getClusterName())).add(CLUSTER_STATE_PATH_TOKEN).add(clusterUUID);
+    }
+
+    public BlobPath getBlobPathForUpload(final AbstractRemoteWritableBlobEntity<T> obj) {
+        BlobPath blobPath = getBlobPathPrefix(obj.clusterUUID());
```

Comment:
We are getting basePath from clusterState repo rather than routing table repo. This can give wrong paths.